aws-amplify/amplify-android
# feat(datastore): Add recoverability improvements (v1)

*Issue #, if available:*
#2201

*Description of changes:*
These are the same changes as #2201 applied to the v1 branch.

*How did you test these changes?*
- Wrote additional unit tests
- Manual testing with emulator
- Simulation of different network conditions via a MITM (e.g. 5XX, 4XX, timeouts, connection resets, etc)

*Documentation update required?*
  - aws-amplify/docs#4986

*General Checklist*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
